### DISCUSSION TITLE
What's the difference between handleBotRequest and handleBrowserRequest in the default entry.server.tsx?

### DISCUSSION THREAD

**Discussion by kotx** [remix-run/remix#5549] [Q&A]
It seems that handleBotRequest uses onAllReady rather than onShellReady, but what's the difference?
It's not covered in [the docs](https://remix.run/docs/en/v1/guides/migrating-react-router-app#creating-server-and-browser-entrypoints), which use a much simpler version of entry.server.tsx.

**Comment by brophdawg11** [comment]
These are React streaming APIs - you can read about them over on the React docs - https://react.dev/reference/react-dom/server/renderToPipeableStream

Specifically, `onAllReady` essentially disables streaming and sends the complete HTML document at the end: 

> onAllReady: A callback that fires when all rendering is complete, including both the [shell](https://react.dev/reference/react-dom/server/renderToPipeableStream#specifying-what-goes-into-the-shell) and all additional [content.](https://react.dev/reference/react-dom/server/renderToPipeableStream#streaming-more-content-as-it-loads) You can use this instead of onShellReady [for crawlers and static generation.](https://react.dev/reference/react-dom/server/renderToPipeableStream#waiting-for-all-content-to-load-for-crawlers-and-static-generation) If you start streaming here, you won’t get any progressive loading. The stream will contain the final HTML.
